### PR TITLE
Rename deploy directory

### PR DESCRIPTION
**Pull request by imcsk8**
The Cluster Version Operator uses a manifests directory,
this patch renames the deploy directory to fit this requirement.

### CODE REVIEW

**Comment by imcsk8** [comment]
PTAL @openshift/sig-network-edge @ironcladlou @Miciah @pravisankar

**Comment by ironcladlou** [comment]
Please update the Dockerfile with this to ensure the manifests directory gets added where it belongs (/manifests)

File: images/cluster-ingress-operator/Dockerfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ COPY . $GOPATH/$CIO_PATH
 RUN cd $GOPATH/$CIO_PATH \
     && make build \
     && cp $GOPATH/$CIO_PATH/cluster-ingress-operator /usr/bin/
+    && cp -rp $GOPATH/$CIO_PATH/manifests /manifests
```
Comment (ironcladlou) [inline images/cluster-ingress-operator/Dockerfile]: I think this should be:

```
ADD manifests/
```

**Comment by ironcladlou** [comment]
Image build is failing...

**Comment by ironcladlou** [comment]
Thanks!

/lgtm

**Pull request merged**